From frederik.deweerdt@xprog.eu  Mon Sep 14 15:18:46 2009
From: Frederik Deweerdt <frederik.deweerdt@xprog.eu>
Date: Mon, 14 Sep 2009 08:52:37 +0000
Subject: Staging: line6: ffzb returns an unsigned integer
To: greg@kroah.com
Message-ID: <20090914085237.GG3113@gambetta>
Content-Disposition: inline



find_first_zero_bit returns a positive value, use it accordingly.

Signed-off-by: Frederik Deweerdt <frederik.deweerdt@xprog.eu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/line6/capture.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/line6/capture.c
+++ b/drivers/staging/line6/capture.c
@@ -26,7 +26,7 @@
 */
 static int submit_audio_in_urb(struct snd_pcm_substream *substream)
 {
-	int index;
+	unsigned int index;
 	unsigned long flags;
 	struct snd_line6_pcm *line6pcm = snd_pcm_substream_chip(substream);
 	int i, urb_size;
@@ -35,7 +35,7 @@ static int submit_audio_in_urb(struct sn
 	spin_lock_irqsave(&line6pcm->lock_audio_in, flags);
 	index = find_first_zero_bit(&line6pcm->active_urb_in, LINE6_ISO_BUFFERS);
 
-	if (index < 0 || index >= LINE6_ISO_BUFFERS) {
+	if (index >= LINE6_ISO_BUFFERS) {
 		spin_unlock_irqrestore(&line6pcm->lock_audio_in, flags);
 		dev_err(s2m(substream), "no free URB found\n");
 		return -EINVAL;
